rubi/rubi/data/sources/market.py

```diff
@@ -1,485 +1,164 @@
-        if end_time:
-            where.append(Offer.timestamp <= end_time)
-
-        # TODO: this is a current limit of the subgrounds library, it does not support querying all as a funtion option, so a large number must be provided
-        # TODO: help a frend, this would be a good pr to add to the subgrounds library :)
-        if where == []:
-            offers = self.rubicon_market_light.Query.offers(first = first)
-        else:
-            offers = self.rubicon_market_light.Query.offers(first = first, where = where)
-
-        field_paths = [
-            offers.id,
-            offers.pay_gem.symbol,
-            offers.pay_amt_formatted,
-            offers.paid_amt_formatted,
-            offers.buy_gem.symbol,
-            offers.buy_amt_formatted,
-            offers.bought_amt_formatted,
-            offers.maker.id,
-            offers.transaction.id,
-            offers.transaction.timestamp,
-            offers.pay_gem.id,
-            offers.buy_gem.id,
-            offers.pay_amt,
-            offers.buy_amt,
-            offers.paid_amt, 
-            offers.bought_amt
-        ]  
-
-        df = self.subgrounds.query_df(field_paths, pagination_strategy=ShallowStrategy) 
-
-        return df
-
-    def get_filled_offers(self, maker = None, taker = None, pair = None, filled = None, cancelled = None, live = None, pay_gem = None, buy_gem = None, start_time = None, end_time = None, first = 1000000000):
-        """returns a dataframe of offers that have received fill with each row corresponding to the take event on the offer. allows for the following filters:
-
-        :param maker: the address of the maker of the offer, defaults to None
-        :type maker: str, optional
-        :param taker: the address of the taker of the offer, defaults to None
-        :type taker: str, optional
-        :param pair: the address of the pair of the offer, defaults to None. direction of the pair matters and corresponds to the array passed in the following order: [pay_gem, buy_gem]. defaults to None
-        :type pair: array, optional
-        :param filled: whether the offer has been filled, defaults to None
-        :type filled: bool, optional
-        :param cancelled: whether the offer has been cancelled, defaults to None
-        :type cancelled: bool, optional
-        :param live: whether the offer is live, defaults to None
-        :type live: bool, optional
-        :param pay_gem: the address of the pay_gem of the offer, defaults to None
-        :type pay_gem: str, optional
-        :param buy_gem: the address of the buy_gem of the offer, defaults to None
-        :type buy_gem: str, optional
-        :param start_time: the start time of the offer, defaults to None. in unix time
-        :type start_time: int, optional
-        :param end_time: the end time of the offer, defaults to None. in unix time
-        :type end_time: int, optional
+    @classmethod
+    def from_network(
```

Comment:
If you follow what I said above then this could be changed to:

```python
    def from_network(
       cls,
        network: Network
    ): 
```

and the `Network` would have a `Subgrounds` instance